src/components/composer/composer/abstract_composer_store.ts

```diff
@@ -556,6 +559,51 @@ export abstract class AbstractComposerStore extends SpreadsheetStore {
     return highlight && highlight.color ? highlight.color : undefined;
   }
 
+  private computeFormulaCursorContext() {
+    if (this.selectionStart === this.selectionEnd) {
+      const parenthesesCodeAtCursor = this.getParenthesesCodeAfterCursor();
+      const previousSymbolAtCursor = [...this.currentTokens]
+        .reverse()
+        .find((t) => parenthesesCodeAtCursor.startsWith(t.parenthesesCode!) && t.type === "SYMBOL");
+
+      if (previousSymbolAtCursor) {
+        for (let i = 0; i < this.currentTokens.length; i++) {
+          this.currentTokens[i].isFormulaCursorContextToken = this.isTokenInCursorFormulaContext(
+            this.currentTokens[i],
+            previousSymbolAtCursor
+          );
+        }
+        return;
+      }
+    }
+
+    for (let i = 0; i < this.currentTokens.length; i++) {
```

Comment:
wouldn't it make more sense to only change the tokens that need to have a different color ?